hexrd/imageseries/stats.py

```diff
@@ -118,31 +122,42 @@ def average_iter(ims, nchunk, nframes=0):
 
 
 def percentile(ims, pctl, nframes=0):
-    """percentile function over frames"""
+    """percentile function over frames
+
+    ims - the imageseries
+    pctl - the percentile
+    nframes - the number of frames to use (default/0 = all)
+    """
     nf = _nframes(ims, nframes)
     return np.percentile(_toarray(ims, nf), pctl, axis=0).astype(np.float32)
 
 
-def percentile_iter(ims, pctl, nchunks,  nframes=0):
+def percentile_iter(ims, pctl, nchunks,  nframes=0, use_buffer=True):
     """iterator for percentile function"""
     nf = _nframes(ims, nframes)
     nr, nc = ims.shape
     stops = _chunk_stops(nr, nchunks)
     r0 = 0
     img = np.zeros(ims.shape)
+    buffer = _alloc_buffer(ims, nf) if use_buffer else None
     for s in stops:
         r1 = s + 1
-        img[r0:r1] = np.percentile(_toarray(ims, nf, (r0, r1)), pctl, axis=0)
-        r0 = s
+        img[r0:r1] = np.percentile(
+            _toarray(ims, nf, rows=(r0, r1), buffer=buffer),
+            pctl, axis=0
+        )
+        r0 = r1
         yield img.astype(np.float32)
```

Comment:
If we are going to always convert the image to `np.float32`, should we just set the dtype as `np.float32` in the call to `np.zeros`?